### PR TITLE
[AppSec] WAF update

### PR DESCRIPTION
**Pull request by ValentinZakharov**
Updated:

- WAF (v1.0.14); 
- Report events (v1.0.0); 
- Rules (v2.1);

### CODE REVIEW

File: dd-java-agent/appsec/appsec.gradle [hunk 1]
```diff
@@ -23,7 +23,7 @@ dependencies {
   api deps.slf4j
   implementation project(':internal-api')
   implementation project(':communication')
-  implementation group: 'io.sqreen', name: 'libsqreen', version: '3.0.2'
+  implementation group: 'io.sqreen', name: 'libsqreen', version: '3.0.3'
```
Comment (bantonsson) [inline dd-java-agent/appsec/appsec.gradle:26]: I assume that this version of `libsqreen` maps to WAF `1.0.14`?

File: dd-java-agent/appsec/src/main/java/com/datadog/appsec/report/EventEnrichment.java [hunk 2]
```diff
@@ -41,102 +39,58 @@
...
-        request.setHost(hostAndPort);
-      } else {
-        request.setHost(hostAndPort.substring(0, hostPosOfColon));
-      }
-    }
     final String uriRaw = appSecCtx.get(KnownAddresses.REQUEST_URI_RAW);
     if (request.getUrl() == null) {
       request.setUrl(buildFullURIExclQueryString(scheme, hostAndPort, uriRaw));
     }
-    if (request.getPort() == null) {
```
Comment (bantonsson) [inline dd-java-agent/appsec/src/main/java/com/datadog/appsec/report/EventEnrichment.java:112]: Are these now extracted in the backend, or has this been removed completely?

File: dd-java-agent/appsec/src/main/java/com/datadog/appsec/report/EventEnrichment.java [hunk 3]
```diff
@@ -153,69 +107,56 @@ public static void enrich(Attack010 attack, IGSpanInfo spanInfo, DataBundle appS
...
 
     Service service = (Service) context.getService();
     if (service == null) {
       service =
           new Service.ServiceBuilder()
-              .withProperty("context_version", "0.1.0")
-              .withProperty("name", Config.get().getServiceName())
-              .withProperty("environment", Config.get().getEnv())
-              .withProperty("version", Config.get().getVersion())
+              .withContextVersion("0.1.0")
```
Comment (bantonsson) [inline dd-java-agent/appsec/src/main/java/com/datadog/appsec/report/EventEnrichment.java:149]: How are we supposed to know about the versions? There's `1.0.0` in some places and `0.1.0` in some. I still think this whole scheme is over-engineered, and brittle.

**Review by bantonsson — COMMENTED** [review]
Have these changes been verified with the system tests?

File: dd-java-agent/appsec/appsec.gradle (hunk 1 above)
Comment (ValentinZakharov) [reply dd-java-agent/appsec/appsec.gradle:26]: yes correct

File: dd-java-agent/appsec/src/main/java/com/datadog/appsec/report/EventEnrichment.java (hunk 3 above)
Comment (ValentinZakharov) [reply dd-java-agent/appsec/src/main/java/com/datadog/appsec/report/EventEnrichment.java:149]: Completely agree, the context versions are a bit confusing. As I understand, basic version was `0.1.0`, but some entities were added (replaced) with new version `1.0.0`. So, now we have for example for events `AppSecEvent100` for current version `1.0.0` and `AppSecEvent010` for pervious version `0.1.0` in parallel. But there are exist non-changed entities which are still `0.1.0` (for instance `Service010`)

File: dd-java-agent/appsec/src/main/java/com/datadog/appsec/report/EventEnrichment.java (hunk 2 above)
Comment (ValentinZakharov) [reply dd-java-agent/appsec/src/main/java/com/datadog/appsec/report/EventEnrichment.java:112]: Yes, in recent version of protocol, port extracts from URI on backend side.

File: dd-java-agent/appsec/appsec.gradle [hunk 4]
```diff
@@ -82,7 +82,7 @@ jsonSchema2Pojo {
 
   // Whether to use the 'title' property of the schema to decide the class name (if not
   // set to true, the filename and property names are used).
-  useTitleAsClassname = false
+  useTitleAsClassname = true
```
Comment (cataphract) [inline dd-java-agent/appsec/appsec.gradle:85]: Seems like you should delete the `doLast` option if you're not relying on custom Java classes for the json schema and only the generated ones now.

**Review by cataphract — APPROVED** [review]

File: dd-java-agent/appsec/src/main/java/com/datadog/appsec/gateway/GatewayBridge.java [hunk 5]
```diff
@@ -89,18 +89,18 @@ public void init() {
...
             // reduced datadog sampling rate.
             spanInfo.setTag(DDTags.MANUAL_KEEP, true);
             spanInfo.setTag("appsec.event", true);
           }
 
-          for (Attack010 attack : collectedAttacks) {
-            EventEnrichment.enrich(attack, spanInfo, ctx);
-            reportService.reportAttack(attack);
+          for (AppSecEvent100 event : collectedEvents) {
+            EventEnrichment.enrich(event, spanInfo, ctx, ctx.isBlocked());
```
[the remark this answers is not in the document]
Comment (ValentinZakharov) [reply dd-java-agent/appsec/src/main/java/com/datadog/appsec/gateway/GatewayBridge.java]: fixed

File: dd-java-agent/appsec/appsec.gradle (hunk 4 above)
Comment (ValentinZakharov) [reply dd-java-agent/appsec/appsec.gradle:85]: removed `doLast`

**Pull request merged**